### DISCUSSION TITLE
How to get the code location of a given asset?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18192] [Q&A] [answered]
Is it possible to get the code location of a given asset? or the "current" one or something? :smile:

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1700551800164649?thread_ts=1700551800.164649&cid=C01U954MEER).*

**Comment by benpankow** [comment, marked as answer]
Yes, it's possible to extract from the asset context:

```python
@asset
def my_asset(context: AssetExecutionContext):
  location_name = (context.get_step_execution_context()
    .dagster_run.remote_job_origin.external_repository_origin
    .code_location_origin.location_name)
```

**Reply by Rhiyo** [reply]
This no longer works in newer versions of dagster.

`external_job_origin` should be `remote_job_origin` now.

**Reply by maximearmstrong** [reply]
@Rhiyo good call, thanks! The example is now updated.
